pkgs/os-specific/linux/lvm2/common.nix

```diff
@@ -82,6 +85,19 @@ stdenv.mkDerivation rec {
     substituteInPlace make.tmpl.in --replace "@systemdsystemunitdir@" "$out/lib/systemd/system"
   '' + lib.optionalString (lib.versionAtLeast version "2.03") ''
     substituteInPlace libdm/make.tmpl.in --replace "@systemdsystemunitdir@" "$out/lib/systemd/system"
+
+    substituteInPlace scripts/blk_availability_systemd_red_hat.service.in \
+      --replace '/usr/bin/true' '${coreutils}/bin/true'
+
+    substituteInPlace scripts/blkdeactivate.sh.in \
+      --replace '/bin/mountpoint' '${util-linux}/bin/mountpoint' \
+      --replace '/bin/umount' '${util-linux}/bin/umount' \
+      --replace '/bin/lsblk' '${util-linux}/bin/lsblk' \
+      --replace '/bin/findmnt' '${util-linux}/bin/findmnt' \
+      --replace '/bin/sort' '${coreutils}/bin/sort' \
+      --replace '/sbin/mdadm' '${if enableMdadm then mdadm else "/run/current-system/sw"}/bin/mdadm' \
```

Comment:
Is it a bad idea to just replace with unqualified search-in-$PATH bare name?

Reply:
It's not a bad idea per-se, but it won't work here, I think.

That is, because the script does stuff like:
```
        if test -f "$LVM"; then
                LVM_AVAILABLE=1
        else
                LVM_AVAILABLE=0
        fi
```
which, now that I look at it, will still break, because `-f` tests for a "regular file" and we have symlinks in `/run/current-system/sw/bin`. More patching needed.

Reply:
Well, stat without explicit preference about symlinks stats the target…